### PR TITLE
don't increment limit for filer non-matches

### PR DESCRIPTION
**Pull request by nlandolfi**
- fixes bug where auth nondetermistically fails (when you have
      more than 1 session, basically, because auth server limits
      its query and filters)

### CODE REVIEW

File: data/query.go [hunk 1]
```diff
@@ -21,7 +21,26 @@ func RecvAll(s DB_QueryClient) ([]*Record, error) {
 	return results, nil
 }
 
-// only checks kind!
+// Use match to tell if a record should pass through this
+// query (i.e., mathches on Kind and passes filters)
```
Comment (viggyfresh) [inline data/query.go]: *matches

File: data/db.go [hunk 2]
```diff
@@ -214,7 +214,7 @@ func (s *server) Query(q *Query, stream DB_QueryServer) error {
 	}
 
 	iter, err := s.DB.Query(data.Kind(q.Kind.String())).
-		Limit(int(q.Limit)).
+		//Limit(int(q.Limit)).
```
Comment (viggyfresh) [inline data/db.go]: no more limit?

Comment (viggyfresh) [inline data/db.go]: nvm just read PR comment.

Comment (nlandolfi) [inline data/db.go]: yep

File: data/query.go (hunk 1 above)
Comment (nlandolfi) [inline data/query.go]: ty

**Pull request merged**